[TensorExpr] Remove obsolete code for handling dynamic shapes from kernel.cpp.

ZolotukhinM opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #36952 [WIP][TensorExpr] kernel.cpp cleanup
* **#36686 [TensorExpr] Remove obsolete code for handling dynamic shapes from kernel.cpp.**
* #36951 Teach IRParser to parse strides along with sizes in a tensor type.

Differential Revision: [D21053305](https://our.internmc.facebook.com/intern/diff/D21053305)

Commit:
[TensorExpr] Remove obsolete code for handling dynamic shapes from kernel.cpp.

[ghstack-poisoned]

Krovatkin reviewed (approved):
do you guys compute contiguity yrself since I haven't added it yet?
:shipit:

Commit:
Update on "[TensorExpr] Remove obsolete code for handling dynamic shapes from kernel.cpp."


Differential Revision: [D21053305](https://our.internmc.facebook.com/intern/diff/D21053305)

[ghstack-poisoned]

Commit:
rebase on "[TensorExpr] Remove obsolete code for handling dynamic shapes from kernel.cpp."


Differential Revision: [D21053305](https://our.internmc.facebook.com/intern/diff/D21053305)

[ghstack-poisoned]

Commit:
Update on "[TensorExpr] Remove obsolete code for handling dynamic shapes from kernel.cpp."


Differential Revision: [D21053305](https://our.internmc.facebook.com/intern/diff/D21053305)

[ghstack-poisoned]

Commit:
Fix test_jit/test_gpu_fusion on "[TensorExpr] Remove obsolete code for handling dynamic shapes from kernel.cpp."


Differential Revision: [D21053305](https://our.internmc.facebook.com/intern/diff/D21053305)

[ghstack-poisoned]

Commit:
Teach IR printer to show strides too. on "[TensorExpr] Remove obsolete code for handling dynamic shapes from kernel.cpp."


Differential Revision: [D21053305](https://our.internmc.facebook.com/intern/diff/D21053305)

[ghstack-poisoned]

Commit:
Update on "[TensorExpr] Remove obsolete code for handling dynamic shapes from kernel.cpp."


Differential Revision: [D21053305](https://our.internmc.facebook.com/intern/diff/D21053305)

[ghstack-poisoned]
